packages/py-api/py_api/controllers/hackathon_teams_controller.py

```diff
@@ -0,0 +1,211 @@
+        move_members_dump: Dict[str, Any] = move_members_model.model_dump()
+        old_team_name = move_members_dump["old_team_name"]
+        new_team_name = move_members_dump["new_team_name"]
+        team_members_to_move = move_members_dump["team_members"]
+
+        old_team = cls.check_if_team_exists(old_team_name)
+        if not old_team:
+            return JSONResponse(
+                content={
+                    "message": f"The team {old_team_name} was not found, please check the spelling!",
+                },
+                status_code=404,
+            )
+
+        new_team = cls.check_if_team_exists(new_team_name)
+        if not new_team:
+            cls.create_team(new_team_name, members_list=team_members_to_move)
+            if content := cls.update_old_team_members_list(old_team, team_members_to_move):
+                return JSONResponse(content={"message": content}, status_code=404)
+
+            return JSONResponse(
+                content={
+                    "message": f"Created a new team with participants {team_members_to_move}",
+                },
+                status_code=201,
+            )
+
+        if content := cls.update_old_team_members_list(old_team, team_members_to_move):
+            return JSONResponse(content={"message": content}, status_code=404)
+
+        cls.update_new_team_members_list(new_team, team_members_to_move)
+
+        return JSONResponse(
+            content={
+                "message": f"Team members {team_members_to_move} moved successfully from team {old_team_name}"
+                           f" to team {new_team_name}",
+            },
+        )
+
+    @classmethod
+    def update_old_team_members_list(
+            cls, old_team: Dict[str, Any],
+            team_members_to_move: List[str],
+    ) -> str | None:
+        old_team_members: List[str] = old_team["team_members"]
+
+        # Remove the passed team_members from their original team
+        for team_member in team_members_to_move:
+            if team_member in old_team_members:
+                old_team_members.remove(team_member)
+            else:
+                return f"Team member with id {team_member} doesn't exist"
+
+        cls.update_team_current_members(
+            old_team["team_name"], old_team_members,
+        )
+        return None
+
+    @classmethod
+    def update_new_team_members_list(cls, new_team: Dict[str, Any], team_members_to_move: List[str]) -> None:
+        new_team_current_members: List[str] = new_team["team_members"]
+
+        # Add the passed team_members to the new team
+        for new_team_member in team_members_to_move:
+            new_team_current_members.append(new_team_member)
+
+        cls.update_team_current_members(
+            new_team["team_name"], new_team_current_members,
+        )
+
+    @classmethod
+    def update_team_current_members(cls, team_name: str, members_list: List[str]) -> None:
```

Comment:
Why break the updating functionality in such small pieces instead of creating one method which updates teams similarly to how you do it in participants.

Get the team and overwrite all fields which match the provisioned values, save it. Reuse wherever you need it. Doesn't make sense to clutter the code in such a way if we don't shoot for really high efficiency.